app/models/miq_provision_virt_workflow.rb

```diff
@@ -359,9 +361,9 @@ def allowed_templates(options = {})
   end
 
   def allowed_template_condition
-    return ["vms.template = ? AND vms.ems_id IS NOT NULL", true] unless self.class.respond_to?(:provider_model)
+    return ["vms.ems_id IS NOT NULL"] unless self.class.respond_to?(:provider_model)
 
-    ["vms.template = ? AND vms.ems_id in (?)", true, self.class.provider_model.pluck(:id)]
+    ["vms.ems_id in (?)", self.class.provider_model.pluck(:id)]
```

Comment:
Does the condition here have to be a string now? (not sure why it was a string in the first place.  Can it be

```suggestion
    {"vms.ems_id" => self.class.provider_model.pluck(:id)}
```

Looking at that I think there's also a way to do a subtable query without making it go back to SQL to get the list of ids.

Reply:
yeah, I can make the first suggestion change.  The second suggestion is good and I wondered if we can do that differently, but ultimately, it's unrelated to excluding deprecated templates so I can do a followup if it is doable.

Reply:
We can do all of it in a follow up, so no need to change now.